vulnerabilities/importers/nvd.py

```diff
@@ -20,150 +20,175 @@
 #  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
 #  Visit https://github.com/nexB/vulnerablecode/ for support and download.
 
-import dataclasses
 import gzip
 import json
 from datetime import date
+from typing import Iterable
 
 import requests
 from dateutil import parser as dateparser
+from django.db.models.query import QuerySet
 
-from vulnerabilities.helpers import create_etag
-from vulnerabilities.importer import Advisory
+from vulnerabilities.helpers import get_item
+from vulnerabilities.importer import AdvisoryData
 from vulnerabilities.importer import Importer
 from vulnerabilities.importer import Reference
 from vulnerabilities.importer import VulnerabilitySeverity
-from vulnerabilities.severity_systems import scoring_systems
-
-BASE_URL = "https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-{}.json.gz"
+from vulnerabilities.improver import Improver
+from vulnerabilities.improver import Inference
+from vulnerabilities.models import Advisory
+from vulnerabilities.severity_systems import SCORING_SYSTEMS
 
 
 class NVDImporter(Importer):
-    def updated_advisories(self):
+    spdx_license_expression = "LicenseRef-scancode-unknown"
+
+    def advisory_data(self):
+        advisory_data = []
         current_year = date.today().year
         # NVD json feeds start from 2002.
         for year in range(2002, current_year + 1):
-            download_url = BASE_URL.format(year)
-            # Etags are like hashes of web responses. We maintain
-            # (url, etag) mappings in the DB. `create_etag`  creates
-            # (url, etag) pair. If a (url, etag) already exists then the code
-            # skips processing the response further to avoid duplicate work
-            if create_etag(data_src=self, url=download_url, etag_key="etag"):
-                data = self.fetch(download_url)
-                yield self.to_advisories(data)
-
-    @staticmethod
-    def fetch(url):
-        gz_file = requests.get(url)
-        data = gzip.decompress(gz_file.content)
-        return json.loads(data)
-
-    def to_advisories(self, nvd_data):
-        for cve_item in nvd_data["CVE_Items"]:
-            if self.is_outdated(cve_item):
```

Comment:
Why removing everything about "outdated"? 
We probably want to avoid these IMHO... but these should be logged alright

Reply:
This is not about outdated data, it's about cutoff time and last run date